ksql-functional-tests/src/main/java/io/confluent/ksql/test/serde/kafka/KafkaSerdeSupplier.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.serde.kafka;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.test.serde.SerdeSupplier;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.connect.data.ConnectSchema;
+import org.apache.kafka.connect.data.Field;
+
+public class KafkaSerdeSupplier implements SerdeSupplier<Object> {
+
+  private final Supplier<LogicalSchema> schemaSupplier;
+
+  public KafkaSerdeSupplier(final Supplier<LogicalSchema> schemaSupplier) {
```

Comment:
I'm not sure I understand why this can't just be `LogicalSchema` instead of `Supplier<LogicalSchema>` - it seems like it shouldn't change.

Reply:
It's an artefact of the way its invoked. There are multiple ways the test `Topic` can be created and the test framework can create a `Topic` without having access to a `LogicalSchema`, only to later overwrite it with one that does have access to one.  Without the use of the `Supplier` the first case would fail unnecessarily.